shm04/react-capstone
# React capstone project - Metrics webapp

## In this pull request I add the following changes: 

- Home page

    - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
    - Along with the category name, you will display some numeric values per category that come from the API.
    - When the user clicks (or taps) on a category item, the application navigates to the details page.
   
- Details page

    - In the details page, the webapp retrieves data from the API to show detailed data in the category.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- Testing requirements

    - Create unit tests for pure functions in your Redux code.
    - Create integration tests for your application using the React Testing Library.
    - You may need to mock the access to the API, so that your tests don't send actual requests.
    - You may need to mock the connection to the Redux Store.

- Technical requirements

    - The project is a single-page application (SPA) built with React and Redux.
    - The data retrieved from the API should be stored in the Redux store.
    - You should filter the data that you retrieve from the API using a Filter stateless component.
    - Every page (the main page and the pages for each item) should have a unique route within the SPA.
    - The project should be deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
    - Add a link to the online version of your app to your README file.
